src/datastore/postgres-store.ts

```diff
@@ -2936,6 +2937,48 @@ export class PgDataStore extends (EventEmitter as { new (): DataStoreEventEmitte
     });
   }
 
+  async getAddressNFTEvent(args: {
+    stxAddress: string;
+    limit: number;
+    offset: number;
+  }): Promise<{ results: AddressNftEventIdentifier[]; total: number }> {
+    return this.query(async client => {
+      const result = await client.query<AddressNftEventIdentifier & { count: string }>(
+        `
+        WITH lastAddressTransfers AS (
+          SELECT MAX(block_height) as max_block_height, asset_identifier, value
+          FROM nft_events
+          WHERE canonical = true AND recipient = $1
+          GROUP BY asset_identifier, value
+        ),
+        lastNFTTransfers AS (
+          SELECT MAX(block_height) as max_block_height, asset_identifier, value
+          FROM lastAddressTransfers FULL JOIN nft_events USING (asset_identifier, value)
+          WHERE canonical = true AND asset_identifier = lastAddressTransfers.asset_identifier
+          GROUP BY asset_identifier, value
+        )
+        SELECT sender, recipient, asset_identifier, value, COUNT(*) OVER() AS count
+        FROM lastNFTTransfers FULL JOIN nft_events USING (asset_identifier, value)
```

Comment:
Looks like we'll need a multicolumn Index on `(asset_identifier, value)` for this query. You can just add 
```ts
pgm.createIndex('nft_events', ['asset_identifier', 'value']);
```
to `src/migrations/1588261750265_nft_events.ts`